### PR TITLE
Percent encode opensearch index name

### PR DESCRIPTION
**Pull request by seankao-az**
### Description
Handle special characters when generating Flint index name by percent-encoding the invalid OpenSearch index name characters, since in some cases, sql queries (DROP, ALTER) are handled over here rather than forwarded to spark.
Same fix in opensearch-spark repo: https://github.com/opensearch-project/opensearch-spark/pull/215

 
### Issues Resolved
* https://github.com/opensearch-project/opensearch-spark/issues/192
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by noCharger** [comment]
It might cause parsing issue. Let's sync up offline.

File: spark/src/test/java/org/opensearch/sql/spark/asyncquery/IndexQuerySpecTest.java [hunk 1]
```diff
@@ -27,9 +27,13 @@
 import org.opensearch.sql.spark.rest.model.LangType;
 
 public class IndexQuerySpecTest extends AsyncQueryExecutorServiceSpec {
```
Comment (dai-chen) [inline spark/src/test/java/org/opensearch/sql/spark/asyncquery/IndexQuerySpecTest.java:29]: could you add simple test in `IndexQuerySpecVacuumTest` too?

**Review by penghuo — DISMISSED** [review]

**Comment by seankao-az** [comment]
> It might cause parsing issue. Let's sync up offline.

special character in index name should be quoted in backticks `` already.

**Review by dai-chen — APPROVED** [review]
Thanks for the changes!

**Review by penghuo — APPROVED** [review]

**Pull request merged**